client/state/reducer-registry/test/index.js

```diff
@@ -0,0 +1,391 @@
+			reducers = combineReducersAndAddLater(
+				{
+					age,
+				},
+				'test',
+				Object.create( null )
+			);
+			const state = reducers( partiallyLoadedAppState, write );
+			expect( state ).to.eql( { age: 20, height: 171 } );
+		} );
+
+		test( 'should not load height, because it is missing a schema', () => {
+			const state = reducers( appState, load );
+			expect( state ).to.eql( { age: 20, height: 160 } );
+		} );
+
+		test( 'should load height into _initialState without a loaded reducer', () => {
+			reducers = combineReducersAndAddLater(
+				{
+					age,
+				},
+				'test',
+				Object.create( null )
+			);
+			const state = reducers( appState, load );
+			expect( state ).to.eql( { age: 20, height: { _initialState: 171 } } );
+		} );
+
+		test( 'should work without any initial reducers', () => {
+			reducers = combineReducersAndAddLater( {}, 'test', Object.create( null ) );
+			const state = reducers( appState, load );
+			expect( state ).to.eql( { age: { _initialState: 20 }, height: { _initialState: 171 } } );
+		} );
+
+		test( 'should validate age', () => {
+			const state = reducers( { age: -5 }, load );
+			expect( state ).to.eql( { age: 0, height: 160 } );
+		} );
+
+		test( 'actions work as expected', () => {
+			const state = reducers( appState, grow );
+			expect( state ).to.eql( { age: 21, height: 172 } );
+		} );
+
+		test( 'nested reducers work on load', () => {
+			const registry = Object.create( null );
+			reducers = combineReducersAndAddLater(
+				{
+					age,
+					height,
+					date,
+				},
+				'person',
+				registry
+			);
+			const nested = combineReducersAndAddLater(
+				{
+					person: reducers,
+					count,
+				},
+				'root',
+				registry
+			);
+			const valid = nested( { person: { age: 22, date: 224 } }, load );
+			expect( valid ).to.eql( {
+				person: { age: 22, height: 160, date: new Date( 224 ) },
+				count: 1,
+			} );
+
+			const invalid = nested( { person: { age: -5, height: 100, date: -5 } }, load );
+			expect( invalid ).to.eql( {
+				person: { age: 0, height: 160, date: new Date( 0 ) },
+				count: 1,
+			} );
+		} );
+
+		test( 'nested reducers work on persist', () => {
+			const registry = Object.create( null );
+			reducers = combineReducersAndAddLater(
+				{
+					age,
+					height,
+					date,
+				},
+				'person',
+				registry
+			);
+			const nested = combineReducersAndAddLater(
+				{
+					person: reducers,
+					count,
+				},
+				'root',
+				registry
+			);
+			const valid = nested( { person: { age: 22, date: new Date( 224 ) } }, write );
+			expect( valid ).to.eql( { person: { age: 22, height: 160, date: 224 }, count: 1 } );
+
+			const invalid = nested( { person: { age: -5, height: 100, date: new Date( -500 ) } }, write );
+			expect( invalid ).to.eql( { person: { age: -5, height: 160, date: -500 }, count: 1 } );
+		} );
+
+		test( 'deeply nested reducers work on load', () => {
+			const registry = Object.create( null );
+			reducers = combineReducersAndAddLater(
```

Comment:
I would add a case that mimics what your PR is intending:
```
const extensions = combineReducersAndAddLater( {}, 'extensions' );
combineReducers( { foo, bar, extensions } );
//...
```

Reply:
Addressed by adding more tests.